test_regress/driver.py

```diff
@@ -2344,6 +2498,72 @@ def fst_identical(self, fn1: str, fn2: str) -> None:
+                self.error(f"Signal {signal_name} doesn't exist in the second object\n")
+
+            other_signal = second.nets[signal_name]
+            if other_signal.width != saif_signal.width:
+                self.error("Incompatible signal width in "
+                           f"{signal_name} {saif_signal.width} != {other_signal.width}\n")
+
+            for bit_index in range(saif_signal.width):
+                signal_bit = saif_signal.bits[bit_index]
+                other_signal_bit = other_signal.bits[bit_index]
+
+                if (signal_bit.high_time != other_signal_bit.high_time
+                        or signal_bit.low_time != other_signal_bit.low_time
+                        or signal_bit.transitions != other_signal_bit.transitions):
+                    self.error("Incompatible signal bit parameters in "
+                               f"{signal_name}[{bit_index}]\n")
+
+        if len(first.child_instances) != len(second.child_instances):
+            self.error(f"Number of child instances doesn't match in {first.scope_name}: "
+                       f"{len(first.child_instances)} != {len(second.child_instances)}")
+
+        for instance_name, instance in first.child_instances.items():
+            if instance_name not in second.child_instances:
+                self.error(f"Instance {instance_name} doesn't exist in the second object\n")
+
+            self.compare_saif_instances(instance, second.child_instances[instance_name])
+
+    def compare_saif_contents(self, first: SAIFParser, second: SAIFParser):
+        """Test if second SAIF file has the same values as the first"""
+
+        if first.duration != second.duration:
+            self.error(f"Duration of trace doesn't match: {first.duration} != {second.duration}")
+
+        if first.divider != second.divider:
+            self.error(f"Dividers don't match: {first.divider} != {second.divider}")
+
+        if first.timescale != second.timescale:
+            self.error(f"Timescale doesn't match: {first.timescale} != {second.timescale}")
+
+        if len(first.top_instances) != len(second.top_instances):
+            self.error("Number of top instances doesn't match: "
+                       f"{len(first.top_instances)} != {len(second.top_instances)}")
+
+        for top_instance_name, top_instance in first.top_instances.items():
+            if top_instance_name not in second.top_instances:
+                self.error(f"Top instance {top_instance_name} missing in other SAIF")
+
+            self.compare_saif_instances(top_instance, second.top_instances[top_instance_name])
+
+    def saif_identical(self, fn1: str, fn2: str) -> None:
+        """Test if two SAIF files have logically-identical contents"""
+        test_result_saif = SAIFParser()
+        test_result_saif.parse(fn1)
+
+        golden_saif = SAIFParser()
+        golden_saif.parse(fn2)
+
+        self.compare_saif_contents(golden_saif, test_result_saif)
```

Comment:
I'm thinking we should move all of the saif compare to a new program, e.g. nodist/verilator_saif_diff that returns 0 success or prints an error.   driver is getting pretty large, and if we add more formats this gets more unsupportable.

Probably we should do likewise with VCD, but that's a later project.